### PR TITLE
Show Storage configuration on admin configuration page

### PR DESCRIPTION
**Pull request by zeripath**
Moves storages to a manager and adds their configuration to the admin page

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: modules/storage/manager.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	"encoding/json"
+	"sort"
+	"sync"
+)
+
+var manager *Manager
+
+// Manager is the storage manager
+type Manager struct {
+	mutex sync.Mutex
```
Comment (lafriks) [inline modules/storage/manager.go:17]: Better is to use RWMutex

Comment (zeripath) [reply modules/storage/manager.go:17]: Fair enough, the mutex is held for only the shortest of times - that's why I didn't bother with the rwmutex but I will switch to it

File: modules/storage/manager.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		}
+	}
+	return manager
+}
+
+// Get will return the storage for the provided name
+func (m *Manager) Get(name string) ObjectStorage {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	return m.storage[name].ObjectStorage
```
Comment (6543) [inline modules/storage/manager.go:57]: what happen if there is no storage for name ...
more specific: can do a ObjectStorage with no values do harmfull things?

**Comment by yardenshoham** [comment]
This is in progress for a while now, I'm closing it to avoid stale pull requests. Please reopen when it's ready for review.

**Pull request closed without merging**